x-pack/test/detection_engine_api_integration/utils/remove_server_generated_properties.ts

```diff
@@ -22,5 +23,9 @@ export const removeServerGeneratedProperties = (
     execution_summary,
     ...removedProperties
   } = rule;
-  return removedProperties;
+  // We're only removing undefined values, so this cast correctly narrows the type
+  return pickBy(removedProperties, (value) => value !== undefined) as Omit<
```

Comment:
We have to remove them because in the switch to `FullResponseSchema`, the mock data now has to specify fields that are `undefined` explicitly (e.g. https://github.com/elastic/kibana/pull/137605/files#diff-d162965f34c11868eafde7e03ca9bfe234c93845bc38bc7c441494b4d7cf5ca9R47-R59), but when we compare the mocks to actual outputs in the integration tests the extra `undefined` fields cause the equality check to fail. 

This should be fixed as a follow up when we make a response type that does not require all fields to be specified, but for now I figured getting everything onto the same base set of schemas should happen first, then we can continue to expand the capabilities of the schema definition system.

Reply:
> but when we compare the mocks to actual outputs in the integration tests the extra undefined fields cause the equality check to fail.

Ah, I see! Thank you.

> This should be fixed as a follow up when we make a response type that does not require all fields to be specified

FWIW I think the problem could be in assertions rather than mocks. Instead of `expect(<actual>).toEqual(<expected>)` we could assert only a subset of the properties that would be relevant to each particular test: `expect(<actual>).toEqual(expect.objectContaining(<expected subset>))`.

Anyway, that would be a separate story to work on...